File: app/src/main/java/com/jameskbride/eventbusmvpdemo/main/MainActivity.java
package com.jameskbride.eventbusmvpdemo.main;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.jameskbride.eventbusmvpdemo.EventBusMVPDemoApplication;

import javax.inject.Inject;


public class MainActivity extends AppCompatActivity {

    @Inject
    MainActivityImpl delegate;

    @Override public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ((EventBusMVPDemoApplication)getApplication()).getApplicationComponent().inject(this);
        delegate.onCreate(savedInstanceState, this);
    }

    @Override public void onResume() {
        super.onResume();
        delegate.onResume(this);
    }
}

File: app/src/main/java/com/jameskbride/eventbusmvpdemo/main/MainActivity.kt
package com.jameskbride.eventbusmvpdemo.main

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.jameskbride.eventbusmvpdemo.EventBusMVPDemoApplication
import javax.inject.Inject


class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var delegate:MainActivityImpl

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        EventBusMVPDemoApplication.applicationComponent.inject(this)
        delegate.onCreate(savedInstanceState, this)
    }

    override fun onResume() {
        super.onResume()
        delegate.onResume()
    }

    override fun onPause() {
        super.onPause()
        delegate.onPause()
    }
}
